src/webhook/mutation/pod_mutator/oneagent_mutation/annotations.go

```diff
@@ -26,7 +27,8 @@ func setInjectedAnnotation(pod *corev1.Pod) {
 
 func getInstallerInfo(pod *corev1.Pod) installerInfo {
 	return installerInfo{
-		flavor:        kubeobjects.GetField(pod.Annotations, dtwebhook.AnnotationFlavor, arch.FlavorMultidistro),
+		arch:          kubeobjects.GetField(pod.Annotations, dtwebhook.AnnotationArch, arch.Arch),
```

Comment:
this annotation has to be removed, since the architecture of the webhook is not necessarily the same as that of the injected pod

Reply:
@chrismuellner thanks for the code review and quick fix. You had a solid point in the case of mixed node architecture environments. The best way is to have the standalone installer / init container handle it.

There is a holdover issue where `FlavorMultidistro` does not exist for s390x or ppc64le. The same issue occurs where we assume so in the webhook, and inject that into the pod initContainer.

So we should probably also set the install flavor to `arch.Flavor` in the standalone installer, as you did for `arch.Arch` in #880.

Reply:
makes sense, we'll do that in a separate PR as well to fix it for the next release already.
Thanks for pointing that out.

Reply:
@chrismuellner That sounds perfect!

I've restored the original source files for the pod mutator webhook from main, reverting any changes to those pkg's.